src/app/main/component/user/components/profile/profile-cards/profile-cards.component.ts

```diff
@@ -22,27 +23,83 @@ export class ProfileCardsComponent implements OnInit, OnDestroy {
   ) {}
 
   ngOnInit() {
-    this.languageSubscription = this.localStorageService.languageBehaviourSubject.subscribe(() => {
+    this.localStorageService.languageBehaviourSubject.pipe(takeUntil(this.destroy$)).subscribe(() => {
+      this.loadHabitFactFromLocalStorage();
       this.getFactOfTheDay();
+      this.checkAndUpdateHabitFact();
     });
+
+    timer(0, this.profileService.oneDayInMillis)
+      .pipe(takeUntil(this.destroy$))
+      .subscribe(() => this.checkAndUpdateHabitFact());
+  }
+
+  loadHabitFactFromLocalStorage(): void {
+    this.habitFactOfTheDay = this.profileService.getHabitFactFromLocalStorage();
   }
 
   getFactOfTheDay(): void {
-    this.profileSubscription = this.profileService
```

Comment:
you removed this.profileSubscription, but did not remove ProfileCardsComponent.spec
please update `should unsubscribe from subscriptions on component destroy`
`component.languageSubscription`
`component.profileSubscription`